python/interpret_community/explanation/explanation.py

```diff
@@ -1666,145 +1669,161 @@ def get_value(x):
-            if param in expldict:
-                local_kwargs[param] = _transform_value_for_load(paramkey, expldict, _metadata)
-        local_explanation = _create_local_explanation(
-            explanation_id=id_value,
-            init_data=_transform_value_for_load('INIT_DATA', expldict, _metadata),
-            classification=_transform_value_for_load('CLASSIFICATION', expldict, _metadata),
-            expected_values=_transform_value_for_load('EXPECTED_VALUES', expldict, _metadata),
-            **local_kwargs)
-        if expldict.get(ExplainParams.GLOBAL_IMPORTANCE_VALUES, None) is not None:
-            # BOTH Local and Global explanation
-            global_kwargs = dict()
-            if getattr(ExplainParams, 'LOCAL_EXPLANATION') in expldict:
-                paramkeys.remove('LOCAL_EXPLANATION')
-            for paramkey in paramkeys:
-                param = getattr(ExplainParams, paramkey)
-                if param in expldict:
-                    global_kwargs[param] = _transform_value_for_load(paramkey, expldict, _metadata)
-            return _create_global_explanation(
-                explanation_id=id_value,
-                init_data=_transform_value_for_load('INIT_DATA', expldict, _metadata),
-                classification=_transform_value_for_load('CLASSIFICATION', expldict, _metadata),
-                expected_values=_transform_value_for_load('EXPECTED_VALUES', expldict, _metadata),
-                local_explanation=local_explanation,
-                **global_kwargs)
-        else:
-            # Local explanation ONLY
-            return local_explanation
-    else:
-        # Global explanation ONLY
-        global_kwargs = dict()
-        for paramkey in paramkeys:
-            param = getattr(ExplainParams, paramkey)
-            if param in expldict:
-                global_kwargs[param] = _transform_value_for_load(paramkey, expldict, _metadata)
-        return _create_global_explanation(
-            explanation_id=id_value,
-            init_data=_transform_value_for_load('INIT_DATA', expldict, _metadata),
-            classification=_transform_value_for_load('CLASSIFICATION', expldict, _metadata),
-            expected_values=_transform_value_for_load('EXPECTED_VALUES', expldict, _metadata),
-            **global_kwargs)
+def load_explanation(path):
+    shared_params = [
+        ExplainParams.EXPECTED_VALUES,
+        ExplainParams.FEATURES,
+        ExplainParams.CLASSES,
+    ]
+    global_params = [
+        ExplainParams.GLOBAL_IMPORTANCE_RANK,
+        ExplainParams.GLOBAL_IMPORTANCE_VALUES,
+        ExplainParams.PER_CLASS_RANK,
+        ExplainParams.PER_CLASS_VALUES
+    ]
+    local_params = [
+        ExplainParams.LOCAL_IMPORTANCE_VALUES
+    ]
+    with open(os.path.join(path, 'explanation_metadata.json'), 'r') as f:
+        metadata = json.load(f)
+        is_global = metadata[ExplainType.GLOBAL]
+        is_local = metadata[ExplainType.LOCAL]
+    if is_local and is_global:
```

Comment:
nit: I think this can be cleaned up a bit more as:
```
if is_local:
  local_kwargs = _get_kwargs(path, shared_params + local_params)
  explanation = _create_local_explanation(**local_kwargs)
if is_global:
  if is_local:
    global_kwargs = _get_kwargs(path, global_params, local_explanation=explanation)
  else:
    global_kwargs = _get_kwargs(path, shared_params + global_params)
  explanation = _create_global_explanation(**global_kwargs)
return explanation
```

Reply:
There are a couple things that would need to be added to the way you're showing it to make it work (tried it out) - either way, it's 11 lines, and I think the original way is a little easier to read